### ISSUE TITLE
[EasyApiToken] Default decoder

### ISSUE THREAD

**Issue by natepage** [eonx-com/easy-monorepo#319]
Right now, to instantiate a decoder, we need to use the decoder factory and pass it the name of the decoder we want to use. Most apps have a "default" decoder they use all the time.

I think it would be cool to have a concept of "default decoder" and if someone has a dependency on the `ApiTokenDecoderInterface` then this package will take care of instantiating the default decoder and inject it without having to use the decoder factory explicitly.

Basically the same way as Doctrine with the `EntityManagerInterface`.

**Comment by egor-dev** [comment]
Good idea. We could introduce `default_decoder` root parameter in the `easy-api-token` config.
As per Manage, this approach will allow us to get rid of the following code in the `AuthServiceProvider`:
```
$this->app->singleton(
    EasyApiTokenDecoderInterface::class,
    function () {
        return $this->app->make(EasyApiTokenDecoderFactoryInterface::class)->build('chain');
    }
);
```

**Comment by natepage** [comment]
@egor-dev Not sure yet how to define the default decoder because in #286 I've implemented the structure to allow apps to define the decoders in PHP and I think that it should become the default way and we should deprecate the framework specific configs.

But yeah it could be as simple as a method on the decoder factory to identify the default decoder.

**Comment by roman-eonx** [comment]
@natepage can we also introduce a default encoder? It will simplify our tests a bit:
```
        $encoder = new JwtTokenEncoder(
            new FirebaseJwtDriver('HS256', 'my-key', 'my-key', ['HS256'], 1)
        );
```
Instead of instantiating the encoder and passing the driver explicitly, we could simply inject `EasyApiTokenEncoderInterface`.

**Comment by natepage** [comment]
@roman-eonx To be really honest with you, the encoders are not really good in general because I've tried to make them generic it would need more thoughts to make it good, and right now the encoders are only used in tests...

So I was thinking about removing them completely.

**Comment by natepage** [comment]
@roman-eonx do you use those encoders somewhere else than in your tests?

**Comment by roman-eonx** [comment]
@natepage nope, only in tests.

**Comment by natepage** [comment]
I've update the PR to add the support for `default_decoder`.

@roman-eonx and it's the same for all projects, that's why I've decided to remove them because like I said they were not all working properly and mostly not used 😄 

Maybe we can find a replacement for it, one that doesn't require that much code and works properly.
